docs/src/main/sphinx/admin/properties-resource-management.rst

```diff
@@ -24,10 +28,21 @@ by the hash tables built during execution, memory used during sorting, etc.
 When the user memory allocation of a query on any worker hits this limit,
 it is killed.
 
+.. warning::
+   :ref:`prop-resource-query-max-total-memory` must be greater than
+   :ref:`prop-resource-query-max-memory`, and
+   :ref:`prop-resource-query-max-memory` must be greater than
+   :ref:`prop-resource-query-max-memory-per-node`. Also, the sum of
+   :ref:`prop-resource-query-max-memory-per-node` and
+   :ref:`prop-resource-memory-heap-headroom-per-node` must be less than the
+   maximum heap size in the JVM on the node. See :ref:`jvm_config`.
```

Comment:
Q: Is this actually enforced somewhere? Or is this just our recommendation.

e.g. I don't see any reason why `query-max-memory` MUST be greater than `query-max-memory-per-node`. Think of a cluster tuned for very high concurrency for example.

The sum of max-per-node + heap-headroom < jvm heap and max-total > max-memory are the only required ones AFAIK.

Reply:
I guess not. I was running on a bad assumption mixed with the old model with system memory.

Reply:
Should this even be a recommendation or should o just remove it?

Reply:
The recommendation is useful but on the other hand user gets explicit error message saying the same thing you say here.

I think it's useful to document though since it's unlikely the constraints will change often and cause docs to go out of date.

Reply:
Sounds good I'll separate it as a note rather than a warning.